From 7b622a3c4e6cbc26a43359e4bde64d9fcf383102 Mon Sep 17 00:00:00 2001
From: jqssun <jqssun+git@googlemail.com>
Date: Tue, 28 Apr 2026 21:10:56 -0700
Subject: [PATCH] support pc


diff --git a/app/src/main/java/com/atharok/btremote/common/utils/Constants.kt b/app/src/main/java/com/atharok/btremote/common/utils/Constants.kt
index 8a39a20..8ce05b8 100644
--- a/app/src/main/java/com/atharok/btremote/common/utils/Constants.kt
+++ b/app/src/main/java/com/atharok/btremote/common/utils/Constants.kt
@@ -43,7 +43,7 @@ const val USE_MOUSE_NAVIGATION_BY_DEFAULT = false
 
 const val KEYBOARD_REPORT_ID = 0x01
 const val REMOTE_REPORT_ID = 0x02
-const val MOUSE_REPORT_ID = 0x03
+const val MOUSE_REPORT_ID = 0x02
 
 val REMOTE_INPUT_NONE = byteArrayOf(0x00, 0x00)
 
