From 4ab2d21a4ad7292f931b3d40c5023caf096e8e32 Mon Sep 17 00:00:00 2001
From: jqssun <jqssun+git@googlemail.com>
Date: Tue, 28 Apr 2026 21:54:17 -0700
Subject: [PATCH] read keys


diff --git a/app/build.gradle.kts b/app/build.gradle.kts
index b927492..01f78eb 100644
--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -1,8 +1,14 @@
+import java.util.Properties
+
 plugins {
     alias(libs.plugins.android.application)
     alias(libs.plugins.kotlin.compose)
 }
 
+val localProps = Properties().apply {
+    rootProject.file("local.properties").takeIf { it.exists() }?.inputStream()?.use { load(it) }
+}
+
 android {
     namespace = "com.atharok.btremote"
     compileSdk {
@@ -24,10 +30,10 @@ android {
 
     signingConfigs {
         create("release") {
-            System.getenv("ANDROID_KEY_STORE_FILE")?.let { storeFile = file(it) }
-            System.getenv("ANDROID_KEY_STORE_PASSWORD")?.let { storePassword = it }
-            System.getenv("ANDROID_KEY_ALIAS")?.let { keyAlias = it }
-            System.getenv("ANDROID_KEY_PASSWORD")?.let { keyPassword = it }
+            localProps.getProperty("storeFile")?.let { storeFile = file(it) }
+            localProps.getProperty("storePassword")?.let { storePassword = it }
+            localProps.getProperty("keyAlias")?.let { keyAlias = it }
+            localProps.getProperty("keyPassword")?.let { keyPassword = it }
         }
     }
 
